build/AzurePipelinesTemplates/ProjectReunion-BuildProject-Steps.yml

```diff
@@ -60,6 +60,12 @@ steps:
       configuration: '$(buildConfiguration)'
       msbuildArgs: '/p:AppxPackageDir="${{ parameters.appxPackageDir }}" /p:AppxBundle=Never /p:AppxSymbolPackageEnabled=false /binaryLogger:$(Build.SourcesDirectory)/${{ parameters.solutionPath }}.$(buildPlatform).$(buildConfiguration).binlog /p:ProjectReunionVersionBuild=$(builddate_yymm) /p:ProjectReunionVersionRevision=$(builddate_dd)$(buildrevision) /p:VCToolsInstallDir="$(VCToolsInstallDir)\" /p:PGOBuildMode=$(PGOBuildMode)'
 
+  - task: CmdLine@2
```

Comment:
I am adding credscan to the pipeline which is going to cause the pipeline to break because of this certificate. That means we are going to have to remove the pfx and the cer from this repo. We can use Azure Key Vault which we have already setup to place a certificate file in the right place for running the tests.

But for running the tests locally, we should set up a script to run before running the tests to create and install the certificate.

Reply:
I have a branch ready that calls into Azure key vault and writes the cert to a file
https://github.com/microsoft/ProjectReunion/compare/user/kythant/RemovePfx

The successful run from it
https://dev.azure.com/ms/ProjectReunion/_build/results?buildId=142063&view=logs&j=f34e22ea-0873-5cfe-9808-56c0b9fb9988

Reply:
Recommend using [azuresigntool](https://github.com/vcsjones/AzureSignTool) and skipping all the file-based operations completely. (Example in production https://github.com/File-New-Project/EarTrumpet/blob/master/.github/workflows/main.yml#L258-L259)

Reply:
Bit offtopic, but how would that be used to sign APPX package? If you're using a wapproj, you currently directly put a certificate thumbprint in the project project and the build system invokes signtool for you.

Reply:
I highly recommended you forget that feature exists and pass in `/p:AppxPackageSigningEnabled=false` when building Packaging projects. Then sign the artifacts manually.

Reply:
@kythant you can't remove MSTest.cer/pfx without breaking all not-packaged inner-loop development. DynDep's tests need signed test packages during development (inner-loop), as will every other developer working on not-packaged tests.

MSTest.cer/pfx shouldn't be used to sign real packages and you may even have alternative solutions for tests executed during the CI/Pipeline but how do you expect inner-loop testing to happen?

Reply:
Pre-build script in the vcxproj to generate a certificate file if it doesn't exist?

Reply:
> Pre-build script in the vcxproj to generate a certificate file if it doesn't exist?

Adding a cert requires admin privilege. 

This is generally something devs do once up front - just like they install VS, the needed Windows SDKs, etc. Requiring an admin prompt for a 1-time thing isn't unreasonable. Requiring it every time you hit F5 is...unwise. We do NOT want devs routinely forced to use an admin prompt to do development. To compound the problem, DynDep doesn't support elevation so running VS elevated is also counter-productive.

Yes it's possible to add a project "IF not certstore.Exists(ReunionMSTest) { GenerateCert(); Run(elevated, "certstore.exe -addcert TrustedPeople blah.cer"); } but that's not free or seamless. It's desirable for inner-loop to be fast and friction-free. Adding 1-time setup work to inner-loop is the wrong direction.

Remember, this is something every non-packaged test will require. Having the build automagically JustDoIt(TM) may be reasonable for CI/Pipeline but not for day to day development.

Reply:
AFAIK, it shouldn't need admin to add it in the user store instead of the machine store, and that's how VS does it (it's able to create and register certs for sideloading MSIX without admin)

Reply:
> that's how VS does it (it's able to create and register certs for sideloading MSIX without admin)

Are you referring to when you debug a VS packaged project? F5?

That calls `packageManager.RegisterPackageAsync(...,DeploymentOptions=DevelopentMode)` resulting in a package with [Package.SignatureKind](https://docs.microsoft.com/uwp/api/windows.applicationmodel.package.signaturekind)=Developer, or more specifically [GetStagedPackageOrigin()](https://docs.microsoft.com/windows/win32/api/appmodel/nf-appmodel-getstagedpackageorigin)=DeveloperUnsigned.

[DeploymentOptions.DevelopmentMode](https://docs.microsoft.com/en-us/uwp/api/Windows.Management.Deployment.DeploymentOptions) has restrictions. Even if you can register a framework package this way (not entirely positive; I'm checking) it's generally not recommended. It can also lead to some odd test problems e.g. one test runs as a user, registers the framework then fails or you're debugging and stop before any cleanup. Tests running as another user (say, you) will fail to register. I'd prefer to make it harder to cause problems down the road, not easier.